### PR TITLE
Voice changer randomizes your name if emp'd

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
If your voice changer mask is emp'd, it generates a random name every message.

## Why It's Good For The Game
Gives a unique effect for the voice changer emp (it currently does nothing), can also be seen as a positive if you want to cause chaos over the radio

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

</details>

## Changelog
:cl:
tweak: voice changer now randomizes your name when emp'd
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
You need failsafe to give a name when it failed to give a name. I.e.) when speces nane file has nothing.

**Comment by Penwin0** [comment]
What Dragon said, have a failsafe if it fails to get a name

File: code/modules/clothing/chameleon.dm [hunk 1]
```diff
@@ -543,7 +543,8 @@
 	permeability_coefficient = 0.01
 	flags_cover = MASKCOVERSEYES | MASKCOVERSMOUTH
 
-	var/vchange = 1
+	var/vchange = TRUE
+	var/rname_cooldown = 0
```
Comment (Archanial) [inline code/modules/clothing/chameleon.dm]: Please use cooldown defines

File: code/modules/mob/living/carbon/human/say.dm [hunk 2]
```diff
@@ -12,6 +12,9 @@
 		var/obj/item/clothing/mask/chameleon/V = wear_mask
 		if(V.vchange && wear_id)
 			var/obj/item/card/id/idcard = wear_id.GetID()
+			if(V.rname_cooldown > world.time)
+				var/randomize = dna.species.random_name(rand(1, 2))
+				return randomize | "John Doe"  // fallback
```
Comment (Archanial) [inline code/modules/mob/living/carbon/human/say.dm]:
```suggestion
				return randomize ?? "John Doe"  // fallback
```

**Review by Archanial — CHANGES_REQUESTED** [review]

**Review by Penwin0 — CHANGES_REQUESTED** [review]
Cooldown defines are used incorrectly, please look at their use in other code

**Comment by etherware-novice** [comment]
yk you could just tell me whats wrong instead of being vauge? i dont see anything done wrong here

File: code/modules/mob/living/carbon/human/say.dm [hunk 3]
```diff
@@ -12,6 +12,9 @@
 		var/obj/item/clothing/mask/chameleon/V = wear_mask
 		if(V.vchange && wear_id)
 			var/obj/item/card/id/idcard = wear_id.GetID()
+			if(COOLDOWN_FINISHED(V, rname_cooldown))
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/say.dm]: Shouldn't it be if the cooldown hasn't finished?

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/modules/clothing/chameleon.dm [hunk 4]
```diff
@@ -560,10 +561,12 @@
 	if(. & EMP_PROTECT_SELF)
 		return
 	chameleon_action.emp_randomise()
+	COOLDOWN_START(src, rname_cooldown, 2 MINUTES)
 
 /obj/item/clothing/mask/chameleon/broken/Initialize(mapload)
 	. = ..()
 	chameleon_action.emp_randomise(INFINITY)
+	rname_cooldown = INFINITY  // mulligan
```
Comment (PowerfulBacon) [inline code/modules/clothing/chameleon.dm:571]: Ironically, EMPing this will fix it, since it will get set to 2 mins instead of infinity.

**Comment by etherware-novice** [comment]
@PowerfulBacon can you reopen

**Pull request closed without merging**